### PR TITLE
Tune CONTRIBUTION

### PR DESCRIPTION
**Pull request by pepe**
As discussed in https://github.com/janet-lang/janet/discussions/563, this is the first draft of the new contribution rules for this repository.

Please comment below if you find anything you do not like or think could be written better.

EJIFG

### CODE REVIEW

**Comment by bakpakin** [comment]
I like the new guidelines and especially the emphasis on making it work with jpm.

**Review by bakpakin — APPROVED** [review]

**Comment by pepe** [comment]
One more thing that comes to mind is the optional `janet-` prefix in the GitHub repository to avoid name clashes for more general package names, e.g., sh, bigint, or set.

**Comment by pepe** [comment]
In #22, I show an example of Tier 3 (in my eyes) package.

**Comment by sogaiu** [comment]
This is perhaps a minor point, but some software is in the public domain, so there is strictly speaking no license IIUC.

It seems a bit weird to say "public domain" or "cc0" in a license field.

**Comment by pepe** [comment]
@sogaiu that is an excellent point. What about then not having `:license` field but point to your pd/cc in the readme and point to it in the PR? For me, sure, it is enough.

**Comment by pepe** [comment]
Anyway, one brilliant bird chirped some more betterment on the text in DM. I will add them today, and if no one objects anything, I will merge this at 18:00 UTC

**Comment by sogaiu** [comment]
@pepe Thanks for the comments about the license.  Perhaps we can discuss some text along the lines you suggested in a future PR.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,36 +1,82 @@
...
 
-Please read this document before making contributions.
+Please read this document before making contributions. If you find anything
+missing of not appropriate, please do not hesitate and create an issue in this
+repository.
 
 ## Adding a Package
 
+If you would like to add a package here, open a pull request with the
+package's name and URL and edit the pkgs.janet file. Plus points for simple
```
Comment (pyrmont) [inline CONTRIBUTING.md:12]:
```suggestion
package's name and URL and edit the `pkgs.janet` file. Plus points for simple
```

Comment (pyrmont) [inline CONTRIBUTING.md:12]:
```suggestion
package's name and URL and edit the pkgs.janet file. Bonus points for simple
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,36 +1,82 @@
...
+descriptions or reasons why you are adding the package in the comment.
+
+These are essential requirements on the package. The
```
Comment (pyrmont) [inline CONTRIBUTING.md:15]:
```suggestion
Below are the key requirements for the package. The
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,36 +1,82 @@
...
+
 * Check if the package exists before adding it twice.
-* The name of the package should usually be the same as that of its
-  git repository, and should be all lower case and in kebab-case.
-  For example, use `my-package` as a package name, not `mypackage` or `MyPackage`.
-* Make sure that your package can be used with `jpm`. This means it
-  should have a file `project.janet` in it's root directory, and should run properly when installed via `jpm install https://some.git-host.org/me/mypackage.git`. Do not add packages that do not work with jpm! If jpm cannot support your package, open an issue in the [Janet](https://github.com/janet-lang/janet) repository and we will see what we can do.
+* Make sure that users can use your package with `jpm`. It must have a file
+  `project.janet` in its root directory and properly run when installed via
+  `jpm install https://some.git-host.org/me/mypackage.git`. Do not add
```
Comment (pyrmont) [inline CONTRIBUTING.md:21]:
```suggestion
  `jpm install https://git-host.example.org/path/to/your/package.git`. Do not add
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,36 +1,82 @@
...
+  packages that do not work with jpm! If jpm cannot support your package,
+  open an issue in this repository, and we will see what we can do.
+* Please be sure that your `project.janet` file contains these required fields:
+  * `:name` with the desired name of the package. The package's name should
+  usually be the same as its git repository and should be all lower case
+  and in kebab-case.
```
Comment (pyrmont) [inline CONTRIBUTING.md:27]:
```suggestion
  usually be the same as its git repository and should be in lower
  kebab-case.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,36 +1,82 @@
...
+      * `my_package`
+      * `MyPackage`
+      * `my package`
+      * `my.package.git`
+      * `org.this.is.not.java.AbstractWidgetFactoryProducer`
+  *  `:author` name of the author of the package. Preferably with email contact
+  in the standard format: `"Josef Pospíšil <josef.pospisil@laststar.eu>"`.
+  * `:description` short informative description of the package, preferably one
+  sentence.
+  * `:license` abbreviation of the License name, e.g., "MIT".
```
Comment (pyrmont) [inline CONTRIBUTING.md:43]:
```suggestion
  * `:license` abbreviation of the license name, e.g., "MIT".
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,36 +1,82 @@
...
+  * `:url` with further information about the package. It could be a repository
+  home page with `README.md` or any other website.
+  * `:repo` from which users could clone code.
+  * `:dependencies` field is optional but must be present if it depends on
+  other packages.
+* Add package License, preferably in the LICENSE file in the root of the
```
Comment (pyrmont) [inline CONTRIBUTING.md:49]:
```suggestion
* Add package license, preferably in a `LICENSE` file in the root of the
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,36 +1,82 @@
...
+  Janet release. Tests are also a great way to explore the package API.
 
-### Good package names
+### Tier 2:
 
-* `my-package`
-* `thing`
-* `thing@bakpakin` (for a fork of the package)
+* example files that are easy to follow and shows usage of the package code.
+* CONTRIBUTING file similar to this.
```
Comment (pyrmont) [inline CONTRIBUTING.md:72]:
```suggestion
* `CONTRIBUTING` file similar to this.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,36 +1,82 @@
...
-### Bad package names
+### Tier 3:
 
-* `mypackage`
-* `my_package`
-* `MyPackage`
-* `my package`
-* `my.package.git`
-* `org.this.is.not.java.AbstractWidgetFactoryProducer`
+* TECH file with details about implementation and inner working of the package.
```
Comment (pyrmont) [inline CONTRIBUTING.md:76]:
```suggestion
* `TECH` file with details about implementation and inner workings of the package.
```

**Review by pyrmont — COMMENTED** [review]
Some suggested wording changes.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,36 +1,82 @@
 # Guidelines for contributing to the Janet Package Listing
```
Comment (pyrmont) [inline CONTRIBUTING.md:1]:
```suggestion
# Guidelines for contributing to the Janet package listing
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,36 +1,82 @@
...
 
-Thanks for taking time to contribute!
+Thanks for taking the time to contribute!
 
-Please read this document before making contributions.
+Please read this document before making contributions. If you find anything
+missing of not appropriate, please do not hesitate and create an issue in this
+repository.
 
 ## Adding a Package
```
Comment (pyrmont) [inline CONTRIBUTING.md:9]:
```suggestion
## Adding a package
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,36 +1,82 @@
...
+  * `:url` with further information about the package. It could be a repository
+  home page with `README.md` or any other website.
+  * `:repo` from which users could clone code.
+  * `:dependencies` field is optional but must be present if it depends on
+  other packages.
+* Add package License, preferably in the LICENSE file in the root of the
+  project.
+
+
+## Optional Package properties
```
Comment (pyrmont) [inline CONTRIBUTING.md:53]:
```suggestion
## Optional package properties
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,36 +1,82 @@
...
-* `my_package`
-* `MyPackage`
-* `my package`
-* `my.package.git`
-* `org.this.is.not.java.AbstractWidgetFactoryProducer`
+* TECH file with details about implementation and inner working of the package.
 
-If you would like to add a package here, just open a pull request with the name and URL of the package and edit the pkgs.janet file.
 
 ## Broken Links
```
Comment (pyrmont) [inline CONTRIBUTING.md:79]:
```suggestion
## Broken links
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (pepe) [reply CONTRIBUTING.md]: Bonus I will go by hand.

**Pull request merged**